### PR TITLE
Fix conf data persistence

### PR DESCRIPTION
**Pull request by lewis-mackay-mf**
Use an initContainer to ensure presence of conf data: if we don't have any at startup time, copy it into the PV from the image.

Signed-off-by: lewis-mackay-mf

### CODE REVIEW

File: helm/idol-nifi/templates/nifi/nifi.yaml [hunk 1]
```diff
@@ -53,6 +54,37 @@ spec:
         securityContext: {{- omit $component.containerSecurityContext "enabled" | toYaml | nindent 10 }}
           readOnlyRootFilesystem: true
         {{- end }}
+      - name: ensure-conf
```
Comment (daniel-lau-mf) [inline helm/idol-nifi/templates/nifi/nifi.yaml:57]: `ensure-conf` init-container can run before wait-for-zookeeper

File: helm/idol-nifi/templates/nifi/nifi.yaml [hunk 2]
```diff
@@ -53,6 +54,37 @@ spec:
...
+        - configMapRef:
+            name: idol-nifi-env
+            optional: false
+        resources:
+          requests:
+            cpu: {{ $component.nifi.resources.requests.cpu }}
+            memory: {{ printf "%dMi" ( $component.nifi.resources.requests.memoryMi | int) }}
+          limits:
+            cpu: {{ $component.nifi.resources.limits.cpu }}
+            memory: {{ printf "%dMi" ( $component.nifi.resources.limits.memoryMi | int) }}
```
Comment (daniel-lau-mf) [inline helm/idol-nifi/templates/nifi/nifi.yaml:69]: Don't really need to apply the same cpu/memory limits for the init container if we don't execute the nifi main

File: helm/idol-nifi/templates/nifi/nifi.yaml [hunk 3]
```diff
@@ -53,6 +54,37 @@ spec:
...
+        securityContext: {{- omit $component.containerSecurityContext "enabled" | toYaml | nindent 10 }}
+          readOnlyRootFilesystem: true
+        {{- end }}
+        command:
+        - sh
+        - "-c"
+        - |
+          if [ -z "$(ls -A -- /mnt/store/nifi/conf)" ]; then
+            echo Copying files into conf directory
+            cp /opt/nifi/nifi-current/conf/* /mnt/store/nifi/conf/
```
Comment (daniel-lau-mf) [inline helm/idol-nifi/templates/nifi/nifi.yaml:80]:
```suggestion
            cp -r /opt/nifi/nifi-current/conf/ /mnt/store/nifi/conf/
````

Comment (lewis-mackay-mf) [reply helm/idol-nifi/templates/nifi/nifi.yaml:80]: This is not correct, as the data ends up in /mnt/store/nifi/conf/conf

**Review by daniel-lau-mf — APPROVED** [review]

**Pull request merged**